src/sonic-yang-mgmt/sonic_yang.py

```diff
@@ -50,7 +51,18 @@ def __init__(self, yang_dir, debug=False):
     def __del__(self):
         pass
 
+    def disable_logging(self):
+        ly.set_log_options(0)
+        self.syslogging_disabled = True
+
+    def enable_logging(self):
+        # ly.LY_LOLOG: If callback is set use it, otherwise just print. If flag is not set, do nothing.
+        ly.set_log_options(ly.LY_LOLOG)
```

Comment:
An alternative solution to always disable libyang logging. Instead we can use the following:
```
        ly.set_log_options(ly.LY_LOSTORE_LAST)
```
or
```
        ly.set_log_options(ly.LY_LOSTORE)
```
Then in all lines where we call any functionality in libyang we print the errors ourselves using:
```
get_ly_errors(ctx)
```

This gives us more fine tuned control over how logs show up. For my case, I only needed a way to disable/enable logging, so do not need to make it more complex.

Reply:
The option suggested in this comment also provide consistency across instances of `sonic_yang`, assume there are multiple instances created, each instance should have control of how it logs. But relying `ly.LY_LOLOG` can cause inconsistencies between instance. If we rely instead on `ly.LY_LOSTORE` or `ly.LY_LOSTORE_LAST` and `get_ly_errors(ctx)` this can only be associated with a single `context` and there is a single `context` per `sonic_yang` object.

For `generic-config-updater` use case we don't manipulate multiple instances of `sonic_yang` at once, so we don't care about this problem. Also manipulating `yang` global state using `ly.set_log_options` can be done anyway by different files, so there is no easy to fix the logging flags consistency issues.